lib/Parser/Parse.cpp

```diff
@@ -9943,7 +9943,8 @@ ParseNodeCase * Parser::ParseCase(ParseNodePtr *ppnodeBody)
 Parse a single statement. Digest a trailing semicolon.
 ***************************************************************************/
 template<bool buildAST>
-ParseNodePtr Parser::ParseStatement()
+ParseNodePtr Parser::ParseStatement(
+    bool isStmtOfWith/* = FALSE*/)
```

Comment:
Can we do this without passing a parameter? The parser maintains a statement stack (see PushStmt), so I think we can inspect that stack to see if we have knopWith on top.